front/app/api/analytics/useAnalytics.ts

```diff
@@ -15,6 +16,16 @@ const useAnalytics = <Response extends BaseResponseData>(
   query: Query,
   onSuccess?: () => void
 ) => {
+  const queryClient = useQueryClient();
+
+  // Call onSuccess if the query is already in the cache
+  useEffect(() => {
+    const queryKey = analyticsKeys.item(query);
+    if (queryClient.getQueryData(queryKey)) {
+      onSuccess && onSuccess();
+    }
+  }, [query, queryClient, onSuccess]);
```

Comment:
Shouldn't we stringify the `query` before putting it in the dependencies? I think now `onSuccess` is going to fire a lot because `query` is not a stable reference

Reply:
It doesn't seem to fire a lot. I tested it with different filters and it is ok - fires once per chart when something changes, as expected.

Reply:
Ok but if I look at the code it will actually refire whenever the component calling `useAnalytics` re-renders... So then it's not really a `onSuccess` function, but more of a 'on success and on re-render' function. Maybe in the very specific situation where you tested it that was not an issue, but if I want to use this hook somewhere else later and I see a prop called `onSuccess` it would be very easy to introduce bugs because I would expect it to only fire on success